### PR TITLE
[Chat][Feature] Emit Redux Action when ChatView is detached

### PR DESCRIPTION
**Pull request by v-loalbert**
## Purpose

*  Emit Redux Action when ChatView is detached

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

### CODE REVIEW

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/container/ChatView.kt [hunk 1]
```diff
@@ -12,20 +12,30 @@ import com.azure.android.communication.ui.chat.locator.ServiceLocator
...
         addView(composeView)
+        count = 0
     }
 
     override fun onAttachedToWindow() {
         super.onAttachedToWindow()
-        ReduxViewModelGenerator(
+        count++
+        if (count >= 1) {
+            dispatch(LifecycleAction.EnterForegroundTriggered())
```
Comment (ahammer) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/container/ChatView.kt:36]: Only needs to be on 1

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/action/LifecycleAction.kt [hunk 2]
```diff
@@ -4,9 +4,9 @@
 package com.azure.android.communication.ui.calling.redux.action
 
 internal sealed class LifecycleAction : Action {
-    class EnterForegroundTriggered : LifecycleAction()
-    class EnterBackgroundTriggered : LifecycleAction()
+    object EnterForegroundTriggered : LifecycleAction()
```
Comment (ahammer) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/action/LifecycleAction.kt:7]: Do we need triggered/succeeded?

We are dispatching after this has happened, it's already "succeeded"

I think EnterForeground, EnterBackground is probably enough.

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/container/ChatView.kt (hunk 1 above)
Comment (v-loalbert) [reply azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/container/ChatView.kt]: fixed

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/action/LifecycleAction.kt (hunk 2 above)
Comment (v-loalbert) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/action/LifecycleAction.kt:7]: fixed

**Review by ahammer — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**